spring-boot/admin/src/test/java/eu/coatrack/admin/controllers/PublicAPIAuthenticationAndAuthorizationViaHeaderTokenTest.java

```diff
@@ -0,0 +1,66 @@
+package eu.coatrack.admin.controllers;
+
+import eu.coatrack.admin.security.CustomUserInfoTokenServices;
+import eu.coatrack.api.ServiceApiDTO;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Request;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class PublicAPIAuthenticationAndAuthorizationViaHeaderTokenTest {
+
+    private static final String AUTHORIZATION_TOKEN = "testToken";
+    private static final String PUBLIC_API_URL = "http://localhost:8080/public-api/services";
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @MockBean
+    private CustomUserInfoTokenServices customUserInfoTokenServicesMock;
+
+    //test to call the public api and get the services
+    @Test
+    public void CallThePublicAPIWithAuthorizationBearerToken_ShouldRetrieveListOfServices() {
+
+        UsernamePasswordAuthenticationToken userAuthentication = new UsernamePasswordAuthenticationToken("exampleCompanyInc",
+                "test", Collections.singleton(new SimpleGrantedAuthority("ROLE_USER")));
+        OAuth2Request request = new OAuth2Request(null, null, null, true, null, null, null, null, null);
+        when(customUserInfoTokenServicesMock.loadAuthentication(AUTHORIZATION_TOKEN)).thenAnswer(invocation -> new OAuth2Authentication(request, userAuthentication));
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBearerAuth(AUTHORIZATION_TOKEN);
+        HttpEntity httpRequest = new HttpEntity(headers);
+
+        ResponseEntity<List<ServiceApiDTO>> getListOfServicesViaPublicAPI = restTemplate.exchange(PUBLIC_API_URL, HttpMethod.GET,
```

Comment:
Please use the Spring WebTestClient instead of deprecated RestTemplate.

Reply:
Done